[evedaktyl] iP

evedaktyl opened the pull request:
# Teletubbi

> "Tasks at bay, mind at play." - evedaktyl
- text-based
- easy to learn
- ~~fast~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/358)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

evedaktyl committed:
Level-0

evedaktyl committed:
Level 1 echo

evedaktyl committed:
Updated Level-1

evedaktyl committed:
Level-2

evedaktyl committed:
Level-3

evedaktyl committed:
Level-4

evedaktyl committed:
Level-4 Text UI Testing

evedaktyl committed:
Level-5 exception handling

evedaktyl committed:
Level-6 delete

evedaktyl committed:
Update task method

evedaktyl committed:
Implement task list saving logic

The chatbot should be able to save tasks in the task list. Let's create a text file that stores task details and read from it when the chatbot is run.

evedaktyl committed:
Merge branch 'branch-Level-7'

evedaktyl committed:
Update Deadline class to store deadlines as LocalDate objects

Spelling out the month makes dates more readable. Let's store dates as LocalDate objects and print them in another format that is more readable.

evedaktyl committed:
Merge branch 'branch-Level-8'

Windofbitter commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,276 @@
+                String[] details = task.split("\\|");
+                String taskType = details[0];
+                Task taskRead = null;
+                switch (taskType) {
```
Coding standard for switch case is followed properly.

Windofbitter commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+     * @return Deadline details in file format.
+     */
+    public String getFileFormat() {
+        return isValidDate
```
I like the way you organize ternary expression, which is very neat and clear, also it align with the coding standard.

Windofbitter commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,276 @@
+        breakLine();
+
+        Task taskAdded;
+        if (input.startsWith("todo")) {
```
Even though the if else seems long, but it is correctly formatted, and I like it

Windofbitter commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,276 @@
+import java.io.IOException;
+
+/**
+ * Represents a chatbot that assists the user with managing a task list.
```
Comment used correctly and it is descriptive

Windofbitter reviewed (commented):
Overall, TLGM, with the coding standards all being followed, especially for the long if else statements. In addition, the switch case format is aligned, as it could be a common mistake to take it for granted the default style.

evedaktyl committed:
Extract classes for more OOP

As the chatbot functionalities get more complex, keeping all functions within the chatbot class will be difficult to manage.
Let's extract out the Ui functionalities, TaskList functionalist and user input parsing functionality into separate classes.

evedaktyl committed:
Update duke.txt file

evedaktyl committed:
Merge branch 'branch-A-moreOOP'

The existing code implements all functionality in the chatbot class, which will be difficult to manage as functionalities grow more complex. Let's extract out the ui
and tasklist management for more OOP.

evedaktyl committed:
Put classes in package duke

Let's organize classes into a suitable package.

evedaktyl committed:
Merge branch 'add-gradle-support'

Use Gradle to automate some of the build tasks of the project.

evedaktyl committed:
Automate project builds using Gradle

evedaktyl committed:
Add JUnit tests for TaskList class and Task class

The TaskList class lacks JUnit tests to validate its functionality.

Without proper unit tests, it's challenging to ensure the correctness and reliability of the various classes.

Let's add JUnit tests to systematically verify the behavior of the various class.

evedaktyl committed:
Merge branch 'branch-A-JUnit'

evedaktyl committed:
Write javadoc for non-private classes and methods

evedaktyl committed:
Edit code to comply with coding style

evedaktyl committed:
Add command to find tasks

evedaktyl committed:
Package as JAR file

evedaktyl committed:
Merge branch 'branch-A-Jar'

evedaktyl committed:
Update JavaDoc

evedaktyl committed:
Merge branch 'branch-A-JavaDoc'

evedaktyl committed:
Merge branch 'branch-A-CodingStandard'

evedaktyl committed:
Merge branch 'branch-Level-9'

lynnetteeee commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,54 @@
+    /**
+     * Sets isDone to false.
+     */
+    public void unmark() {
```
Since this method is related to the previous `markAsDone()` method, it would be good if it is named similarly, maybe `markAsUndone()` !

lynnetteeee commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+        }
+    }
+
+    public TaskList find(String keyword) {
```
Do include a header comment for this method :D Thereafter, can consider adding details to the method name as well, to improve readability and understanding of your code!

lynnetteeee commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,45 @@
+    /**
+     * Prints a greeting to the user.
+     */
+    public void greet() {
```
Small detail but since the method mainly displays/prints  like the `showMessage()` method, perhaps can consider grouping these methods together and give them similar naming, e.g. `showGreetings()` or `showWelcomeMessage()` !

lynnetteeee commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,40 @@
+        assertEquals(t.size(), 0);
+    }
+
+    @Test void delete_invalidIndex_exceptionThrown() {
```
Good naming of test method names with the 3-part format!! :D

lynnetteeee commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+     * @return Task of index i in the list.
+     * @throws DukeException If the index specified is invalid.
+     */
+    public Task get(int i) throws DukeException {
```
More specific naming could be used, e.g. `getTaskFromIndex(int index)` as part of using the method name itself to explain what your code does, for enhanced readability!

lynnetteeee reviewed (commented):
Overall LGTM, good job!! Just need to fix some of the naming :D

evedaktyl committed:
Add checkstyle configuration

evedaktyl committed:
Merge branch 'branch-A-CheckStyle'

evedaktyl committed:
Add JavaFX GUI functionality

evedaktyl committed:
Merge branch 'branch-Level-10'

evedaktyl committed:
Update style and configure gradle for JavaFX

evedaktyl committed:
Add assert feature

Mark and unmark functions assume the task's status icon is updated correctly.

Using assertions helps us to ensure that the status icon is updated correctly whenever a task is marked or unmarked.

evedaktyl committed:
Extract out parsing functions

Parsing function is lengthy and difficult to read.

Let's extract out the parsing of various commands to simplify the main parser.

evedaktyl committed:
Merge pull request #2 from evedaktyl/branch-A-Assertions

Add assert feature

evedaktyl committed:
Merge branch 'master' into branch-A-CodeQuality

evedaktyl committed:
Merge pull request #3 from evedaktyl/branch-A-CodeQuality

Branch a code quality

evedaktyl committed:
Remove unused imports

evedaktyl committed:
Add flexibility for find command

Find command retrieves tasks with descriptions that contain the exact keyword.

Let's introduce some flexibility by comparing strings using Levenshtein Distance to show tasks that are similar enough to the keyword.

evedaktyl committed:
Merge branch 'branch-BCD-Extension'

evedaktyl committed:
Update javadocs for find method

evedaktyl committed:
Add flexibility to find command

evedaktyl committed:
Replace duke instances with teletubbi

evedaktyl committed:
Implement storage save after each call

evedaktyl committed:
Add stage title

evedaktyl committed:
Remove unused imports

evedaktyl committed:
Update README.md

evedaktyl committed:
Fix Ui container height

evedaktyl committed:
Add help command

evedaktyl committed:
Fix style

evedaktyl committed:
Update README.md
